server/src/main/java/io/druid/server/coordination/BatchDataSegmentAnnouncer.java

```diff
@@ -111,11 +103,26 @@ public void announceSegment(DataSegment segment) throws IOException
             if (availableZNode.getCount() >= config.getSegmentsPerNode()) {
               availableZNodes.remove(availableZNode);
             }
-
             done = true;
+          } else {
+            availableZNodes.remove(availableZNode);
```

Comment:
Yes, I was debating about this too. I remove it because most announcements will have a similar size, unless there are significant schema changes across batches.